### DISCUSSION TITLE
Prettier plugin that sorts markdown tables

### DISCUSSION THREAD

**Discussion by SevenOutman** [prettier/prettier#14364] [Show and tell]
I just made this Prettier plugin [prettier-plugin-sort-markdown-table](https://github.com/SevenOutman/prettier-plugin-sort-markdown-table) that sorts markdown tables according to their first columns.

![Kapture 2023-02-15 at 16 34 22](https://user-images.githubusercontent.com/8225666/218974951-2b53b191-e426-4a3a-a553-17c1dd7b56df.gif)

It's handy when I'm [documenting my React components about their props](https://rsuitejs.com/components/button/#props), and wants to keep the table in alphabetic order.

Hope it also helps you. :)

**Comment by fisker** [comment]
Great tool 👍

**Comment by fisker** [comment]
[This line](https://github.com/SevenOutman/prettier-plugin-sort-markdown-table/blob/2f2f47dc847ef0b3f26d8670cc5a7470a3bb3c6b/src/index.ts#L31) seems not correct to me, the first column can have leading space, can be shorter than other rows, should get the first column trimed then sort.

**Reply by SevenOutman** [reply]
Thanks for reviewing! That's right the leading spaces should be considered. I was thinking of `text` to be already formatted by Prettier for some reason 😲 .

**Reply by SevenOutman** [reply]
It's fixed in [v1.0.2](https://github.com/SevenOutman/prettier-plugin-sort-markdown-table/releases/tag/v1.0.2) now.
